### PR TITLE
fix: try treeless clone (ENG-48)

### PR DESCRIPTION
**Pull request by DecathectZero**
tested here:
https://github.com/voiceflow/creator-app/pull/8442

shallow clones have a lot of limitations for comparisons and getting tags etc, however a treeless clone actually fetches ALL commits and all tags, just not blobs / trees:
https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/

In this PR I simply renamed `checkout_shallow_clone.yml` to `checkout_clone.yml` and now there is a option for `clone-type: treeless`. It defaults to the exact same `shallow` code so nothing else should be affected 👍.

For the `update-track` flow, I've also added in treeless instead of CircleCI's `- checkout`.

### CODE REVIEW

File: src/commands/track/update_track.yml [hunk 1]
```diff
@@ -95,7 +95,8 @@ steps:
   - when:
       condition: << parameters.checkout >>
       steps:
-        - checkout # special step to check out source code to working directory.
+        - checkout_clone:
+            clone_type: treeless
```
Comment (DecathectZero) [inline src/commands/track/update_track.yml:99]: we are changing here from a traditional circleCI clone to a treeless one.

**Comment by edison-vflow** [comment]
Hi @DecathectZero 

Great work !

There were errors detected last time across different repos, so we just need to do due diligence and ensure we are all good.
This is what we did last time to ensure we didnt break anything.
Errors were picked up and resolved in this first round of testing.
Even after first round of testing, there were issues we still had to resolve, with `libs` and design `repos` 

So we cant release these changes until we test across all the below repos.


* https://docs.google.com/spreadsheets/d/1K3MIztu0hxk6CB2AFxeonu9FqLGFQEQL5-pllRbywyo/edit?gid=0#gid=0
* We need to do at least one pass where we do a `bors try` to all the repos as there are some nuances to testing some of those repos

The gotchas and some things that were discovered are documented in [this](https://www.notion.so/voiceflow/Shallow-clone-across-all-repos-9e4f6402bcbe4f3f925f583c61c0ec82) notion page 

* We also need to test database-cli as a special case as its got a workflow that runs on tags after a merge to master has occurred.

So until we have bors tried all repos and done a latest tag testing on database-cli, we cant merge the changes only based on the fact that it works on creator-app

**Review by edison-vflow — CHANGES_REQUESTED** [review]
Hi @DecathectZero 

Great work !

There were errors detected last time across different repos, so we just need to do due diligence and ensure we are all good.
This is what we did last time to ensure we didnt break anything.
Errors were picked up and resolved in this first round of testing.
Even after first round of testing, there were issues we still had to resolve, with `libs` and design `repos` 

So we cant release these changes until we test across all the below repos.


* https://docs.google.com/spreadsheets/d/1K3MIztu0hxk6CB2AFxeonu9FqLGFQEQL5-pllRbywyo/edit?gid=0#gid=0
* We need to do at least one pass where we do a `bors try` to all the repos as there are some nuances to testing some of those repos

The gotchas and some things that were discovered are documented in [this](https://www.notion.so/voiceflow/Shallow-clone-across-all-repos-9e4f6402bcbe4f3f925f583c61c0ec82) notion page 

* We also need to test database-cli as a special case as its got a workflow that runs on tags after a merge to master has occurred.

So until we have bors tried all repos and done a latest tag testing on database-cli, we cant merge the changes only based on the fact that it works on creator-app

**Review by xavidop — APPROVED** [review]

**Comment by DecathectZero** [comment]
@edison-vflow so the key thing I think to test for is the individual git commands that each repo uses and see if they still work when you do a fresh treeless clone - that's what I've been trying with the render-config. For example it uses [this](https://github.com/voiceflow/orb-common/blob/20118383e57de7d070c438e76528417036cf500c/src/commands/utils/get_changed_files.yml#L22):
```
FILE_CHANGES="$(git diff --name-status --no-commit-id -r $BASE...$CIRCLE_SHA1)"
```
and I've ran that command locally and verified that it works with a treeless clone.

Tags should not be an issue on treeless clones because all tags are contained - only trees and blobs needs to be re-fetched.

**Pull request merged**